go-controller/pkg/node/gateway_hybrid.go

```diff
@@ -0,0 +1,342 @@
+		ChassisID:   chassisID,
+		InterfaceID: ifaceID,
+		MACAddress:  macAddress,
+		IPAddresses: ips,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return func() error {
+		// Program cluster.GatewayIntf to let non-pod traffic to go to host
+		// stack
+		if err := addDefaultConntrackRulesHybrid(n.name, bridgeName, uplinkName, n.stopChan); err != nil {
+			return err
+		}
+		return nil
+	}, nil
+}
+
+// since we share the host's k8s node IP, add OpenFlow flows
+// -- to steer the NodePort traffic arriving on the host to the OVN logical topology and
+// -- to also connection track the outbound north-south traffic through l3 gateway so that
+//    the return traffic can be steered back to OVN logical topology
+// -- to also handle unDNAT return traffic back out of the host
+func addDefaultConntrackRulesHybrid(nodeName, gwBridge, gwIntf string, stopChan chan struct{}) error {
+	// the name of the patch port created by ovn-controller is of the form
+	// patch-<logical_port_name_of_localnet_port>-to-br-int
+	localnetLpName := gwBridge + "_" + nodeName
+	patchPort := "patch-" + localnetLpName + "-to-br-int"
+	// Get ofport of patchPort, but before that make sure ovn-controller created
+	// one for us (waits for about ovsCommandTimeout seconds)
+	ofportPatch, stderr, err := util.RunOVSVsctl("wait-until", "Interface", patchPort, "ofport>0",
+		"--", "get", "Interface", patchPort, "ofport")
+	if err != nil {
+		return fmt.Errorf("failed while waiting on patch port %q to be created by ovn-controller and "+
+			"while getting ofport. stderr: %q, error: %v", patchPort, stderr, err)
+	}
+
+	// Get ofport of physical interface
+	ofportPhys, stderr, err := util.RunOVSVsctl("get", "interface", gwIntf, "ofport")
+	if err != nil {
+		return fmt.Errorf("failed to get ofport of %s, stderr: %q, error: %v",
+			gwIntf, stderr, err)
+	}
+
+	// replace the left over OpenFlow flows with the FLOOD action flow
+	_, stderr, err = util.AddFloodActionOFFlow(gwBridge)
+	if err != nil {
+		return fmt.Errorf("failed to replace-flows on bridge %q stderr:%s (%v)", gwBridge, stderr, err)
+	}
+
+	nFlows := 0
+	if config.IPv4Mode {
+		// table 0, packets coming from pods headed externally. Commit connections
+		// so that reverse direction goes back to the pods.
+		_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
+			fmt.Sprintf("cookie=%s, priority=100, in_port=%s, ip, "+
+				"actions=ct(commit, exec(load:0x1->NXM_NX_CT_LABEL),, zone=%d), output:%s",
+				defaultOpenFlowCookie, ofportPatch, config.Default.ConntrackZone, ofportPhys))
+		if err != nil {
+			return fmt.Errorf("failed to add openflow flow to %s, stderr: %q, "+
+				"error: %v", gwBridge, stderr, err)
+		}
+		nFlows++
+
+		// table 0, packets coming from external. Send it through conntrack and
+		// resubmit to table 1 to know the state of the connection.
+		_, stderr, err = util.RunOVSOfctl("add-flow", gwBridge,
+			fmt.Sprintf("cookie=%s, priority=50, in_port=%s, ip, "+
```

Comment:
Can we add a table=0 here to make it explicit? also below for the v6 case

Reply:
yeah